### PR TITLE
CVAE - version 2

### PR DESCRIPTION
**Pull request by jodoin**

### CODE REVIEW

File: article/_posts/2019-05-09-CVAE.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    venue:   "NeurIPS 2015"
+pdf: "https://papers.nips.cc/paper/5775-learning-structured-output-representation-using-deep-conditional-generative-models"
+---
+
+
+<center><img src="/deep-learning/images/CVAE/sc007.png" width="700"></center>
+
+## Summary
+
+Sohn et al. propose two different (but very similar) conditional VAE which they call CVAE and Gaussian stochastic NN (GSNN).  Their goal is to make structured prediction from corrupted input data.  Their overall model is illustated in the previous image.
```
Comment (AntoineTheb) [inline article/_posts/2019-05-09-CVAE.md]: illustated -> illustrated

File: article/_posts/2019-05-09-CVAE.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+For the CVAE, the aim is to generate an output $$y$$ (could be an image or a segmentation as will be shown in the results section) conditioned on $$z$$ and $$x$$ : $$p_{\theta}(y\vert x,z)$$.  The objective changes to the following conditional log-likelihood:
+
+<center><img src="/deep-learning/images/CVAE/sc001.png" width="500"></center>
+
+where $$z^l=g_{\phi}(x,y,\epsilon), \epsilon \sim N(0,1)$$
+
+
+Overall, the CVAE consists of three networks : the recongition network $$q_{\psi}(z\vert x,y)$$, the prior network $$p_\theta(z\vert x)$$ and the generation network $$p_\theta(y\vert x,z)$$.  Interestingly, their implementation can be seen as a recurrent model.
+
+The authors mention the existance of a gap between training and testing of the CVAE.  Specifically, at testing time, $$z$$ is drawn from the prior $$p_\theta(z\vert x)$$, but at training time, the recognition network $$q_{\psi}(z\vert x,y)$$ is used. To make prediction during training and testing the same, they set $$q_\psi(z\vert x,y)=q\theta (z\vert x)$$. This resulted into the following objective
```
Comment (AntoineTheb) [inline article/_posts/2019-05-09-CVAE.md]: existance -> existence

File: article/_posts/2019-05-09-CVAE.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+This model is called GSNN. Finally, the objectives of both models are combined in a weighted sum.
+
+<center><img src="/deep-learning/images/CVAE/sc004.png" width="300"></center>
+
+
+
+## Results
+
+Segmentation results on currupted input data are quite convincing.
```
Comment (AntoineTheb) [inline article/_posts/2019-05-09-CVAE.md]: currupted -> corrupted

**Comment by jhlegarreta** [comment]
Just to close the loop and try to propose a solution that solves the rebasing issues in #349 , from ones own `master`
```
git checkout -b CVAE origin/CVAE
git rebase -i HEAD~20
```

(even if we do not need all 20), then drop all those @jodoin did in the branch except the one with the message `add CVAE file in the new article directory`, and I'd dare to say that push-forcing would make the trick:
https://github.com/vitalab/vitalab.github.io/commit/1187dfd150cab5e303123cf4859dd8abc6783c06

That seems to only contain the files @jodoin intended to add, and in the right location.

HTH.

In any case, I'll close the ancient branch CVAE (https://github.com/vitalab/vitalab.github.io/branches).

File: article/_posts/2019-05-09-CVAE.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+---
+layout: review
+title: "Learning Structured Output Representation using Deep Conditional Generative Models"
+tags: deep-learning VAE Segmentation
```
Comment (jhlegarreta) [inline article/_posts/2019-05-09-CVAE.md]: Not sure whether tags are case-insensitive.

**Comment by jodoin** [comment]
> Images are referenced to the `deep-learning` folder which no longer exists. Changing the references to point to the `article` folder should fix the issue.

done

File: article/_posts/2019-05-09-CVAE.md (hunk 4 above)
Comment (jodoin) [reply article/_posts/2019-05-09-CVAE.md]: done

**Comment by jhlegarreta** [comment]
Antoine's comments have been addressed. Merging.

File: article/_posts/2019-05-09-CVAE.md (hunk 2 above)
Comment (jhlegarreta) [reply article/_posts/2019-05-09-CVAE.md]: Done in 6fef65b.

File: article/_posts/2019-05-09-CVAE.md (hunk 1 above)
Comment (jhlegarreta) [reply article/_posts/2019-05-09-CVAE.md]: Done in 6fef65b.

File: article/_posts/2019-05-09-CVAE.md (hunk 3 above)
Comment (jhlegarreta) [reply article/_posts/2019-05-09-CVAE.md]: Done in 6fef65b.

**Pull request merged**